apps/heft/src/plugins/TypescriptPlugin/TypescriptBuilder.ts

```diff
@@ -0,0 +1,1029 @@
+      ? require(this._configuration.tslintToolPath)
+      : undefined;
+
+    this._terminal.writeLine(`Using TypeScript version ${ts.version}`);
+    if (tslint) {
+      this._terminal.writeLine(`Using TSLint version ${tslint.Linter.VERSION}`);
+    }
+
+    ts.performance.enable();
+
+    const measureTsPerformance: PerformanceMeasurer = <TResult extends object | void>(
+      measurementName: string,
+      fn: () => TResult
+    ) => {
+      const beforeName: string = `before${measurementName}`;
+      ts.performance.mark(beforeName);
+      const result: TResult = fn();
+      const afterName: string = `after${measurementName}`;
+      ts.performance.mark(afterName);
+      ts.performance.measure(measurementName, beforeName, afterName);
+      return {
+        ...result,
+        duration: ts.performance.getDuration(measurementName),
+        count: ts.performance.getCount(beforeName)
+      };
+    };
+
+    const measureTsPerformanceAsync: PerformanceMeasurerAsync = async <TResult extends object | void>(
+      measurementName: string,
+      fn: () => Promise<TResult>
+    ) => {
+      const beforeName: string = `before${measurementName}`;
+      ts.performance.mark(beforeName);
+      const resultPromise: Promise<TResult> = fn();
+      const result: TResult = await resultPromise;
+      const afterName: string = `after${measurementName}`;
+      ts.performance.mark(afterName);
+      ts.performance.measure(measurementName, beforeName, afterName);
+      return {
+        ...result,
+        duration: ts.performance.getDuration(measurementName)
+      };
+    };
+
+    if (this._configuration.watchMode) {
+      await this._runWatch(ts, measureTsPerformance);
+    } else {
+      await this._runBuild(ts, tslint, measureTsPerformance, measureTsPerformanceAsync);
+    }
+  }
+
+  public async _runWatch(ts: ExtendedTypescript, measureTsPerformance: PerformanceMeasurer): Promise<void> {
+    //#region CONFIGURE
+    const { duration: configureDurationMs, tsconfig, compilerHost } = measureTsPerformance(
+      'Configure',
+      () => {
+        const _tsconfig: Typescript.ParsedCommandLine = this._loadTsconfig(ts);
+        const _compilerHost: TWatchCompilerHost = this._buildWatchCompilerHost(ts, _tsconfig);
+        return {
+          tsconfig: _tsconfig,
+          compilerHost: _compilerHost
+        };
+      }
+    );
+    this._terminal.writeVerboseLine(`Configure: ${configureDurationMs}ms`);
+    //#endregion
+
+    this._validateTsconfig(ts, tsconfig);
+
+    ts.createWatchProgram(compilerHost);
+
+    return new Promise(() => {
+      /* never terminate */
+    });
+  }
+
+  public async _runBuild(
+    ts: ExtendedTypescript,
+    tslint: typeof Tslint | undefined,
+    measureTsPerformance: PerformanceMeasurer,
+    measureTsPerformanceAsync: PerformanceMeasurerAsync
+  ): Promise<void> {
+    // Ensure the cache folder exists
+    this._fileSystem.ensureFolder(this._configuration.buildCacheFolder);
```

Comment:
Is there an easy way to clear the cache?

Reply:
Deleting the cache folder is the only way to clear it right now. I can add a flag to the `clean` verb to also delete the cache.